air/src/stack/stack_manipulation/mod.rs

```diff
@@ -10,16 +13,20 @@ pub mod tests;
 // ================================================================================================
 
 /// The number of unique transition constraints in stack manipulation operations.
-pub const NUM_CONSTRAINTS: usize = 2;
+pub const NUM_CONSTRAINTS: usize = 50;
 
 /// The degrees of constraints in individual stack manipulation operations.
 pub const CONSTRAINT_DEGREES: [usize; NUM_CONSTRAINTS] = [
     // Given it is a degree 7 operation, 7 is added to all the individual constraints
     // degree.
-
-    // There are two unique constraints in SWAP stack manipulation operation in which the first
-    // element in the current frame is swapped with the second element.
-    8, 8,
+    8, // 1 constraint for PAD operation
+    8, 8, 8, 7, 7, 8, 7, 8, 8, 8, 8, 8, 8, // 13 constraints for DUPn and MOVUPn operations
+    8, 8, // 2 constraints in SWAP operations
+    8, 8, 8, 8, 8, 8, 8,
+    8, // 8 constraints for SWAPWX operations including 8 constraints of SWAPDW operation
+    8, 8, 8, 8, 8, 8, 8, 8, // 8 constraints for SWAPDW operations
+    8, 8, 8, 8, 8, 8, 8, // 7 constraints for MOVDNn operations
+    8, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, // 11 constraints for CSWAP and CSWAPW operations
```

Comment:
Some of the constraint degrees are lower than I expected. For things like the first constraint in `CSWAP` and `CSWAPW` I understand why it is happening (both have a common prefix). But I'm a bit puzzled by degree 7 constraints for `DUPn` and `MOVUPn` - as far as I can tell, they don't share a common prefix. Could you double check and see if we can explain this?